Tools/build/check_extension_modules.py

```diff
@@ -281,6 +289,50 @@ def list_module_names(self, *, all: bool = False) -> set[str]:
             names.update(WINDOWS_MODULES)
         return names
 
+    def generate_stdlib_info(self, config_path: str | None = None) -> None:
+
+        disabled_modules = {modinfo.name for modinfo in self.modules
+                           if modinfo.state in (ModuleState.DISABLED, ModuleState.DISABLED_SETUP)}
+        missing_modules = {modinfo.name for modinfo in self.modules
+                          if modinfo.state == ModuleState.MISSING}
+        na_modules = {modinfo.name for modinfo in self.modules
+                     if modinfo.state == ModuleState.NA}
+
+        config_messages = {}
+        if config_path:
+            try:
+                with open(config_path, encoding='utf-8') as f:
+                    config_messages = json.load(f)
+            except (FileNotFoundError, json.JSONDecodeError) as e:
+                logger.error("Failed to load distributor config %s: %s", config_path, e)
+
+        default_messages = {
+            **{name: f"Windows-only standard library module '{name}' was not found"
```

Comment:
A better way to add quotes to strings is `repr`:
```suggestion
            **{name: f"Windows-only standard library module {name!r} was not found"
```

It might be good to match this message to "Unsupported platform", e.g.
`Unsupported platform for Windows-only standard library module {name!r}`